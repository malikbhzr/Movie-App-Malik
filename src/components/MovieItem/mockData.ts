export const MockItem = {
    Title: 'Lego Marvel Super Heroes', 
    Year: '2013', imdbID: 'tt2620204', 
    Type: 'game', 
    Poster: 'https://m.media-amazon.com/images/M/MV5BOTA5ODA2NTI2M15BMl5BanBnXkFtZTgwNTcxMzU1MDE@._V1_SX300.jpg',
}