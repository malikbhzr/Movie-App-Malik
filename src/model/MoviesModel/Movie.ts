

export interface Movie {
        Title: string;
        Year: string;
        imdbID: string;
        Type: string;
        Poster: string;
}

export interface MovieList {
    Search: Array<Movie>;
    totalResults: number;
    Response: string;
};