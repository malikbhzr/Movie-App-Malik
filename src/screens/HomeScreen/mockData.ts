export const MockList =
    {
        movies:{
            Search: [{
                Title: "Captain Marvel",
                Year: "2019",
                imdbID: "tt4154664",
                Type: "movie",
                Poster: "https://m.media-amazon.com/images/M/MV5BMTE0YWFmOTMtYTU2ZS00ZTIxLWE3OTEtYTNiYzBkZjViZThiXkEyXkFqcGdeQXVyODMzMzQ4OTI@._V1_SX300.jpg"
            }, {
                Title: "Captain Marvel",
                Year: "2019",
                imdbID: "tt4154664",
                Type: "movie",
                Poster: "https://m.media-amazon.com/images/M/MV5BMTE0YWFmOTMtYTU2ZS00ZTIxLWE3OTEtYTNiYzBkZjViZThiXkEyXkFqcGdeQXVyODMzMzQ4OTI@._V1_SX300.jpg"
            }, {
                Title: "Marvel One-Shot: All Hail the King",
                Year: "2014",
                imdbID: "tt3438640",
                Type: "movie",
                Poster: "https://m.media-amazon.com/images/M/MV5BZGFkMTZkMDQtNzM4Yy00YWEwLTkzOWEtZTMyNDRlNmJhYWJhXkEyXkFqcGdeQXVyNTE1NjY5Mg@@._V1_SX300.jpg"
            }, {
                Title: "Marvel One-Shot: Item 47",
                Year: "2012",
                imdbID: "tt2247732",
                Type: "movie",
                Poster: "https://m.media-amazon.com/images/M/MV5BMjNlMzAxNmQtOGEwZi00NTEyLWI0NWYtMTlhNmE2YTA3ZDVhXkEyXkFqcGdeQXVyNTE1NjY5Mg@@._V1_SX300.jpg"
            }, {
                Title: "Marvel One-Shot: Agent Carter",
                Year: "2013",
                imdbID: "tt3067038",
                Type: "movie",
                Poster: "https://m.media-amazon.com/images/M/MV5BZDIwZTM4M2QtMWFhYy00N2VmLWFlMjItMzI3NjBjYTc0OTMxXkEyXkFqcGdeQXVyNTE1NjY5Mg@@._V1_SX300.jpg"
            }, {
                Title: "Marvel One-Shot: A Funny Thing Happened on the Way to Thor's Hammer",
                Year: "2011",
                imdbID: "tt2011109",
                Type: "movie",
                Poster: "https://m.media-amazon.com/images/M/MV5BYmVlYTg3N2QtMWM2OS00YWQyLWI2M2MtMDc0ZjBkZjk1MTY3XkEyXkFqcGdeQXVyNTE1NjY5Mg@@._V1_SX300.jpg"
            }, {
                Title: "Marvel One-Shot: The Consultant",
                Year: "2011",
                imdbID: "tt2011118",
                Type: "movie",
                Poster: "https://m.media-amazon.com/images/M/MV5BNGE4YjU5MDAtYzYzMC00M2RlLTk0NDgtNDU1MjgyMGI0MjI3XkEyXkFqcGdeQXVyNTE1NjY5Mg@@._V1_SX300.jpg"
            }, {
                Title: "Lego Marvel Super Heroes",
                Year: "2013",
                imdbID: "tt2620204",
                Type: "game",
                Poster: "https://m.media-amazon.com/images/M/MV5BOTA5ODA2NTI2M15BMl5BanBnXkFtZTgwNTcxMzU1MDE@._V1_SX300.jpg"
            }, {
                Title: "Adventures of Captain Marvel",
                Year: "1941",
                imdbID: "tt0033317",
                Type: "movie",
                Poster: "https://m.media-amazon.com/images/M/MV5BNjg0NTk3NjUyNF5BMl5BanBnXkFtZTgwNDQ5MjM1MjE@._V1_SX300.jpg"
            }, {
                Title: "Marvel: Ultimate Alliance",
                Year: "2006",
                imdbID: "tt0814207",
                Type: "game",
                Poster: "N/A"
            }],
        },
        totalResults: 151,
        Response: "True"
    }