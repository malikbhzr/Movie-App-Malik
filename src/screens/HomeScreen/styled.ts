import styled from 'styled-components/native';

export const Title = styled.Text`
    font-size: 21px;
    color: #FFFFFF;
    margin-left: 18px;
`;