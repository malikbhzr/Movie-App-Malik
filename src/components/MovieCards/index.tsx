import React from 'react';
import { FlatList } from 'react-native';
import MovieItem from '../MovieItem';
import { MovieList } from '../../model/MoviesModel/Movie';

interface MovieListProps {
  movieList: MovieList;
  nav: any
}

export default function MovieCards({
  movieList,
  nav
}: MovieListProps) {
  return (
      <FlatList
        horizontal
        showsHorizontalScrollIndicator={false}
        keyExtractor={(item: any) => item.imdbID}
        data={movieList}
        renderItem={(movie) => ( <MovieItem nav={nav} movie={movie.item} /> )}
      />
  );
}