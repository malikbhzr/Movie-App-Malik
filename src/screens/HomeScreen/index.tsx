import React, {useEffect, useState} from 'react';
import {ScrollView, SafeAreaView, View, ActivityIndicator} from 'react-native';
import {useSelector, useDispatch} from 'react-redux';
import AppAction from '../../redux/action/AppAction';
import MovieCards from '../../components/MovieCards';
import {Title} from './styled';
import {SearchBar} from 'react-native-elements';
import SplashScreen from 'react-native-splash-screen';

const HomeScreen = ({navigation}: any) => {
  const movieData = useSelector(state => state);
  const dispatch = useDispatch();
  const [searchMovie, setSearchMovie] = useState(null);
  useEffect(() => {
    dispatch(AppAction.getMovies());
    SplashScreen.hide();
  }, []);

const HomeScreen = ({ navigation }: any) => {
    const movieData = useSelector(state => state);
    const dispatch = useDispatch();
    const [ searchMovie, setSearchMovie ] = useState(null);
    useEffect(() => {
        dispatch(AppAction.getMovies());
    }, []);


    const showMovieList = () => {
        const { navigate } = navigation;
        if(movieData && movieData.appData){
            const { Search } = movieData.appData.movies;
            return (
                <ScrollView>
                    <Title marginTop={20}>Movies on Theatre</Title>
                    <MovieCards nav={navigate} width={230} height={340} textWidth={230} movieList={Search} borderRadius={10} />
                    <Title marginTop={0}>Coming Soon</Title>
                    <View style={{ marginTop: 10}}>
                        <MovieCards nav={navigate} width={104} height={154} textWidth={230} movieList={Search} borderRadius={6} />
                    </View>
                    <View style={{ marginTop: -40}}>
                        <MovieCards nav={navigate} width={104} height={154}  movieList={Search} borderRadius={6} />
                    </View>
                </ScrollView>
            )
        }
        return <ActivityIndicator />
    }
    return <ActivityIndicator />;
  };

  return (
    <SafeAreaView style={{flex: 1, backgroundColor: '#1d1e28'}}>
      <SearchBar placeholder="Type Here..." />
      {showMovieList()}
    </SafeAreaView>
  );
};

export default HomeScreen;
