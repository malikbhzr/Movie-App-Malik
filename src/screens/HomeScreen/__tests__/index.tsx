import React from 'react';
import HomeScreen from '../';
import { render, waitForElement } from 'react-native-testing-library';
import { Provider } from 'react-redux'
import { createStore } from 'redux';
import rootReducer from '../../../redux/reducer/ReducerFactory';
import MovieCards from '../../../components/MovieCards';
import { MovieList } from '../../../model/MoviesModel/Movie';
import AppAction from '../../../redux/action/AppAction';
import { MockList } from '../mockData';
// import SplashScreen from 'react-native-splash-screen';

function renderWithRedux(ui: any, { initialState, store = createStore(rootReducer, initialState) }: any = {}) {
    return {
      ...render(<Provider store={store}>{ui}</Provider>),
      store,
    };
  }

  

describe("Home screen", () => {
    beforeEach(() => { 
        // SplashScreen.hide()
        // console.warn('api', AppAction.getMovies())
        
    });
    let navigation = jest.isMockFunction(null);
    let movieList: MovieList;

    it('render home screen', () => {
        let  wrapper = renderWithRedux(<HomeScreen navigation={navigation} />);
    
        expect(wrapper).toMatchSnapshot();
      });


    it('render home screen title', () => {
        const { queryByText } = renderWithRedux(<HomeScreen navigation={navigation} />);
    
        expect(queryByText('Movies on Theatre')).not.toBeNull();
        expect(queryByText('Coming Soon')).not.toBeNull();
      });

      it('render movies list', () => {
        const cardWrapper = renderWithRedux(<MovieCards nav={navigation} movieList={movieList} />);

        expect(cardWrapper).toMatchSnapshot();
    
      });

      it('render movie cards with props', async () => {
        const { queryByText, debug } = renderWithRedux(<MovieCards nav={navigation} movieList={movieList} />);

        expect(queryByText('Movies on Theatre')).toBeNull();
    
      });
});