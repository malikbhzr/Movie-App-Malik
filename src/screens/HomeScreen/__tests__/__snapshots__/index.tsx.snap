// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home screen render home screen 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#1d1e28",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#393e42",
        "borderBottomColor": "#000",
        "borderBottomWidth": 1,
        "borderTopColor": "#000",
        "borderTopWidth": 1,
        "padding": 8,
      }
    }
  >
    <View
      style={
        Object {
          "paddingHorizontal": 0,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#303337",
            "borderBottomWidth": 0,
            "borderColor": "#86939e",
            "borderRadius": 3,
            "flexDirection": "row",
            "minHeight": 30,
            "overflow": "hidden",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "marginLeft": 8,
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
              underlayColor="white"
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#86939e",
                      "fontSize": 18,
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="iconIcon"
              >
                î¢¶
              </Text>
            </View>
          </View>
        </View>
        <TextInput
          allowFontScaling={true}
          editable={true}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder="Type Here..."
          placeholderTextColor="#86939e"
          platform="default"
          rejectResponderTermination={true}
          style={
            Object {
              "alignSelf": "center",
              "color": "#86939e",
              "flex": 1,
              "fontSize": 18,
              "marginLeft": 10,
              "minHeight": 40,
            }
          }
          testID="searchInput"
          underlineColorAndroid="transparent"
          value=""
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "marginLeft": 15,
              "marginRight": 8,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
  <RCTScrollView>
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 21,
              "marginLeft": 18,
            },
          ]
        }
      >
        Movies on Theatre
      </Text>
      <RCTScrollView
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={true}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        showsHorizontalScrollIndicator={false}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View />
      </RCTScrollView>
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 21,
              "marginLeft": 18,
            },
          ]
        }
      >
        Coming Soon
      </Text>
      <RCTScrollView
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={true}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        showsHorizontalScrollIndicator={false}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View />
      </RCTScrollView>
      <RCTScrollView
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={true}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        showsHorizontalScrollIndicator={false}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View />
      </RCTScrollView>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;

exports[`Home screen render movies list 1`] = `
<RCTScrollView
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={true}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  showsHorizontalScrollIndicator={false}
  stickyHeaderIndices={Array []}
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View />
</RCTScrollView>
`;
